go-controller/pkg/network-controller-manager/node_network_controller_manager_test.go

```diff
@@ -163,5 +166,89 @@ var _ = Describe("Healthcheck tests", func() {
 				Expect(execMock.CalledMatchesExpected()).To(BeTrue(), execMock.ErrorDesc)
 			})
 		})
+
+	})
+
+	Context("verify cleanup of deleted networks", func() {
+		var (
+			staleNetID uint   = 100
+			nodeName   string = "worker1"
+			nad               = ovntest.GenerateNAD("bluenet", "rednad", "greenamespace",
+				types.Layer3Topology, "100.128.0.0/16", types.NetworkRolePrimary)
+			netName      = "bluenet"
+			netID        = 3
+			v4NodeSubnet = "10.128.0.0/24"
+			v6NodeSubnet = "ae70::66/112"
+			testNS       ns.NetNS
+			fakeClient   *util.OVNClientset
+		)
+
+		BeforeEach(func() {
+			testNS, err = testutils.NewNS()
+			Expect(err).NotTo(HaveOccurred())
+			v1Objects := []runtime.Object{}
+			fakeClient = &util.OVNClientset{
+				KubeClient: fake.NewSimpleClientset(v1Objects...),
+			}
+		})
+
+		AfterEach(func() {
+			Expect(testNS.Close()).To(Succeed())
+			Expect(testutils.UnmountNS(testNS)).To(Succeed())
+		})
+
+		It("check vrf devices are cleaned for deleted networks", func() {
+			config.OVNKubernetesFeature.EnableNetworkSegmentation = true
```

Comment:
do you need to do Expect(config.PrepareTestConfig()).To(Succeed()) before these tests to make sure config is reset ?

Reply:
noticing data race at 
```
2024-08-06T06:38:46.9252241Z   github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config.PrepareTestConfig()
2024-08-06T06:38:46.9254912Z       /home/runner/work/ovn-kubernetes/ovn-kubernetes/go-controller/pkg/config/config.go:622 +0x3d
```
between `egressfirewall_test.go:253` and `model_client.go:468`, is this a known bug in the test ?

https://productionresultssa8.blob.core.windows.net/actions-results/34957402-aa62-456e-b36b-b0b72fe9809c/workflow-job-run-8cb1fa99-737a-573f-da94-d1c896261bcd/logs/job/job-logs.txt?rsct=text%2Fplain&se=2024-08-06T07%3A37%3A10Z&sig=hPreZqSTiaGFHPuem6Kb9%2Bpg4x2wGyPu4E40m%2FK%2F2Ig%3D&ske=2024-08-06T17%3A39%3A57Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2024-08-06T05%3A39%3A57Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2024-05-04&sp=r&spr=https&sr=b&st=2024-08-06T07%3A27%3A05Z&sv=2024-05-04

Reply:
I saw a similar data race a few days ago:
https://github.com/ovn-org/ovn-kubernetes/pull/4583#issuecomment-2267883251

I don't think it has anything to do with your PR. I'll try to investigate/fix it. Filed https://github.com/ovn-org/ovn-kubernetes/issues/4592